### PR TITLE
Add region deletion

### PR DESCRIPTION
**Pull request by pindge**
- add region deletion for new explorer introduced region table
- delete product_ref from `source_product_refs` and `derived_product_refs`
- tested and can confirm it is working.
- update doc

### CODE REVIEW

**Review by alexgleith — APPROVED** [review]

File: odc-product-delete/delete_odc_product_explorer.sql [hunk 1]
```diff
@@ -28,6 +38,57 @@ SELECT count(*) FROM cubedash.product WHERE name = :'product_name';
 
 DELETE FROM cubedash.product WHERE name = :'product_name';
 
+--
+-- delete lineage bond product
```
Comment (jeremyh) [inline odc-product-delete/delete_odc_product_explorer.sql:42]: What's this section intending to do? It looks like its deleting all products that are upstream or downstream of the given product... are we sure that's what we want? 

I think it'll probably fail? (its not deleting their regions etc. either)

Comment (alexgleith) [reply odc-product-delete/delete_odc_product_explorer.sql:42]: Yeah, I don't think we should be considering automatically deleting lineage products, if that's what "lineage bond" means.

Deleting a product that has a dependency should fail, fast. And deleting a dependency of a deleted product can be a second step. Automating this stuff if fraught with danger.

Comment (jeremyh) [reply odc-product-delete/delete_odc_product_explorer.sql:42]: Agreed.

We probably only need an update statement to remove the deleted product from any `derived_products_ref` arrays (ie. it's own source products). If it has any derived products itself it should fail.

**Comment by pindge** [comment]
```
UPDATE cubedash.product 
SET    derived_product_refs = Array_remove(derived_product_refs, 
                                     (SELECT id 
                                                                  FROM 
                                     cubedash.product 
                                                                  WHERE 
                                     (SELECT id 
                                      FROM   agdc.dataset_type 
                                      WHERE  NAME = :'product_name') = 
                                     ANY ( derived_product_refs ))); 
```

File: odc-product-delete/delete_odc_product_explorer.sql (hunk 1 above)
Comment (pindge) [reply odc-product-delete/delete_odc_product_explorer.sql]: thanks everyone, i thought about and i didn't think too much about it, its been changed to remove that linkage from `source_product_refs` and `derived_product_refs`

**Pull request merged**